daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -489,12 +489,11 @@ private[lf] object SBuiltin {
         args: util.ArrayList[SValue],
         machine: Machine[Q],
     ): Control.Value = {
-      val coid = getSContractId(args, 0).coid
-      machine match {
-        case _: PureMachine | _: ScenarioMachine =>
-          Control.Value(SOptional(Some(SText(coid))))
-        case _: UpdateMachine =>
-          Control.Value(SValue.SValue.None)
+      if (machine.isUpdateMachine) {
```

Comment:
what is the problem with pattern machine ?

Reply:
It requires that `Machine` is covariant in `Q`.

Reply:
It is not really the pattern matching by itself that requires the covariance but more accurately the disjunctive pattern. 

```
 machine match {
        case _: UpdateMachine =>
           Control.Value(SValue.SValue.None)
         case _ => 
}
``` 

just work fine, it is clearer and make the method `isUpdateMachine` useless.

Reply:
Do you know the reason for this?
Or is it just  _Scala_ :shrug: